src/commands/publish/preview/mod.rs

```diff
@@ -18,7 +19,63 @@ struct Preview {
     pub id: String,
 }
 
-pub fn preview(
+fn preview_with_auth(
+    project: &Project,
+    user: &GlobalUser,
+    method: Result<HTTPMethod, failure::Error>,
+    body: Option<String>,
+) -> Result<(), failure::Error> {
+    let create_address = format!(
+        "https://api.cloudflare.com/client/v4/accounts/{}/workers/scripts/{}/preview",
+        project.account_id, project.name
+    );
+
+    let client = http::auth_client(user);
+
+    commands::build(&project)?;
+
+    let script_upload_form = publish::build_script_upload_form(project)?;
+
+    log::info!("💩");
+    let res = client
+        .post(&create_address)
+        .multipart(script_upload_form)
+        .send()?
+        .error_for_status();
+
+    let text = &res?.text()?;
```

Comment:
So, I don't think that this is an *inherently* bad line, but it is a little punctuation heavy. I'd move the first `?` to be on line 44 above, and then move the `&` to the call on line 50.